tsdb/head.go

```diff
@@ -327,7 +387,7 @@ func (h *Head) PostingsCardinalityStats(statsByLabelName string) *index.Postings
 }
 
 // NewHead opens the head block in dir.
-func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, opts *HeadOptions) (*Head, error) {
+func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, opts *HeadOptions, stats *HeadStats) (*Head, error) {
```

Comment:
Can head just give us the stats with some `GetStats()`? Why we need to compose it and create from outside? It's just same instance always and we don't need to be preffiled (we can make head being the owner of initialising this object)

Reply:
We don't get the DB object with the head back until after `Open` has returned, and thereby the WAL replay has been completed.